lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -555,26 +524,17 @@ ArrayAttr TypeLoweringVisitor::filterAnnotations(
       retval.push_back(SubAnnotationAttr::get(ctxt, newFieldID, annotation));
       continue;
     }
-    if (isDontTouch) {
+    if (Annotation(opAttr).getClass() ==
+        "firrtl.transforms.DontTouchAnnotation") {
+      // If this is a non-local DontTouchAnnotation then lower it to a symbol
```

Comment:
I guess, it need not be a non-local DontTouchAnnotation, `needsSym=true` can be triggered even if a local `DontTouchAnnotation` is found!

Reply:
There is an assumption that any local `DontTouchAnnotation` is already represented as a symbol which this comment is assuming. I'll update this to indicate that this comment assumes that.

Reply:
I thought even local `DontTouchAnnotation` on subfields are still preserved as `SubAnno` on the field, and not lowered to symbols!